symengine/ruby/ext/symengine/ruby_basic.c

```diff
@@ -5,23 +5,17 @@ void cbasic_free(void *ptr){
     basic_free(basic_ptr);
 }
 
-VALUE cbasic_alloc(VALUE klass){
-    VALUE obj;
-    basic_struct *struct_ptr;
-
-    obj = Data_Make_Struct(klass, basic_struct, NULL, cbasic_free, struct_ptr);
-
-    struct_ptr->data = NULL;
+void cbasic_free_heap(void *ptr) {
+    basic_struct *basic_ptr = ptr;
+    basic_free_heap(basic_ptr);
+}
 
-    return obj;
+VALUE cbasic_alloc(VALUE klass){
+    basic_struct *struct_ptr = basic_new_heap();
+    return Data_Wrap_Struct(klass, NULL, cbasic_free_heap, struct_ptr);
 }
 
 VALUE cbasic_init(VALUE self){
-    basic_struct *this;
-
-    Data_Get_Struct(self, basic_struct, this);
-    basic_init(this);
-
     return self;
 }
```

Comment:
This function is empty now --- is that intended? Does it still need to be declared? It's only used in
```
     rb_define_method(c_basic, "initialize", cbasic_init, 0);
```